src/arch/mod.rs

```diff
@@ -1,3 +1,9 @@
+//! This module implements "platform agnosticity". No matter what platform you're trying
+//! to get goOSe working on, all of the conditions defined here must be met. On top of
```

Comment:
> all of the conditions defined here must be met

Which ones? I tough of making an arch trait but this will be in another pr with some refactor

Reply:
For now it's just the external symbols, but I'm assuming that we'll have a trait and other conditions at some point

Reply:
Ok for me then